### PR TITLE
Add Custom onboarding pages

### PR DESCRIPTION
**Pull request by Zishan-7**
- Added custom onboarding pages integrated with clerk SDK
- `react-hook-form` is used for form validation.

### CODE REVIEW

File: src/onboarding/Login.jsx [hunk 1]
```diff
@@ -1,7 +1,65 @@
...
+      });
+    } finally {
+      dispatch({ isLoading: false });
+    }
+  };
+
+  const handleGoogleLogin = async () => {
+    await signIn.authenticateWithRedirect({
+      strategy: 'oauth_google',
+      redirectUrl:
```
Comment (Zishan-7) [inline src/onboarding/Login.jsx:57]: Need to update the redirect URL with the Nethermind account redirect URL (need to setup in clerk dashboard)

File: src/onboarding/Register.jsx [hunk 2]
```diff
@@ -1,7 +1,164 @@
...
+        if (e.meta.paramName === 'code') {
+          dispatch({ verificationError: e.longMessage });
+        }
+      });
+    } finally {
+      dispatch({ isLoading: false });
+    }
+  };
+
+  const handleGoogleLogin = async () => {
```
Comment (vincenthongzy) [inline src/onboarding/Register.jsx:86]: does this work? I have no response when testing locally

File: src/onboarding/Login.jsx [hunk 3]
```diff
@@ -12,44 +70,97 @@ export default function Login() {
...
+          isInvalid={!!errors.email}
+          label="Email"
+          name="email"
+          placeholder="Enter your email"
+          size="sm"
+          variant="bordered"
+          {...register('email', {
+            required: 'Email is required',
+            pattern: {
+              value: RegExp(/^[^\s@]+@[^\s@]+\.[^\s@]+$/),
```
Comment (vincenthongzy) [inline src/onboarding/Login.jsx:100]: Hmm emails are notoriously hard to create a regex for. This is definitely too simple

Comment (vincenthongzy) [reply src/onboarding/Login.jsx:100]: https://stackoverflow.com/a/201378

Comment (vincenthongzy) [reply src/onboarding/Login.jsx:100]: Is there a battle-tested library for these

File: src/onboarding/Register.jsx [hunk 4]
```diff
@@ -1,7 +1,164 @@
...
+    handleSubmit,
+    getValues,
+    setError,
+    clearErrors,
+    reset,
+    formState: { errors }
+  } = useForm();
+  const { isLoaded, signUp, setActive } = useSignUp();
+  const [state, dispatch] = useMutativeReducer(reduceState, {
+    isLoading: false,
```
Comment (vincenthongzy) [inline src/onboarding/Register.jsx:21]: hm is it necessary `isLoading` and `isLoaded` together?

File: tailwind.config.js [hunk 5]
```diff
@@ -62,7 +72,8 @@ export default {
             outline: '#293041',
             primary: '#cad7f9',
             secondary: '#ffcc80',
-            success: '#7ccb69'
+            success: '#7ccb69',
+            danger: '#d24646'
```
Comment (vincenthongzy) [inline tailwind.config.js:76]: `nit`:  why not re-use our `error` variants?

File: src/onboarding/Register.jsx [hunk 6]
```diff
@@ -1,7 +1,164 @@
-import { Link } from 'react-router-dom';
-import { SignUp } from '@clerk/clerk-react';
```
Comment (vincenthongzy) [inline src/onboarding/Register.jsx:2]: curious, is it alot of work to customize the `SignUp` form from clerk ?

File: src/onboarding/Register.jsx (hunk 2 above)
Comment (Zishan-7) [reply src/onboarding/Register.jsx:86]: It works at my end. Are you using the correct clerk key?

File: src/onboarding/Register.jsx (hunk 4 above)
Comment (Zishan-7) [reply src/onboarding/Register.jsx:21]: isLoaded is the status for Clerk instance. IsLoading is for showing the loading spinner on the button. Both have separate functions.

File: src/onboarding/Register.jsx (hunk 6 above)
Comment (Zishan-7) [reply src/onboarding/Register.jsx:2]: Not a lot of work. But Clerk pre-built UI does not provide the Terms & conditions checkbox So we had to develop the UI from scratch.

File: src/onboarding/Register.jsx (hunk 2 above)
Comment (vincenthongzy) [reply src/onboarding/Register.jsx:86]: yes works now

File: src/onboarding/Register.jsx (hunk 4 above)
Comment (vincenthongzy) [reply src/onboarding/Register.jsx:21]: It's confusing to have two variables with the same name, could you perhaps rename `isLoading` to something else? Not a blocker of course, but good to have

File: tailwind.config.js (hunk 5 above)
Comment (Zishan-7) [reply tailwind.config.js:76]: Carlos asked to use this color 😬

File: src/onboarding/Register.jsx (hunk 4 above)
Comment (Zishan-7) [reply src/onboarding/Register.jsx:21]: renamed isLoaded to isClerkLoaded

File: tailwind.config.js (hunk 5 above)
Comment (vincenthongzy) [reply tailwind.config.js:76]: Did you let him know there is already something similar with our `error` variants?

**Review by vincenthongzy — APPROVED** [review]

**Comment by Zishan-7** [comment]
As discussed with Clem, hiding google login till oAuth is setup.

**Pull request merged**